### PR TITLE
Update CDC queries for cdc_prev and deletes

### PR DESCRIPTION
**Pull request by kathancox**
Fixes DOC-7460, DOC-7422

This PR updates the CDC queries page:

- Adds an example of how to capture message values for delete events.
- Adds a note that delete messages will only emit `NULL` except for the primary key

### CODE REVIEW

**Review by amruss — APPROVED** [review]
LGTM

**Comment by amruss** [comment]
@kathancox from https://cockroachlabs.atlassian.net/browse/DOC-7422, I think we still need to address this part somewhere in the docs: "need to use with diff with event_op() to get proper behavior. When using with diff, we will never say that something is an UPSERT, only INSERT or UPDATE." 

For the composite key piece, I think its still worth adding a note that changes to a column that is part of the composite key will look like a delete, then an insert (this is true for all changefeeds but I think worth pointing out here as well)

**Comment by kathancox** [comment]
> @kathancox from https://cockroachlabs.atlassian.net/browse/DOC-7422, I think we still need to address this part somewhere in the docs: "need to use with diff with event_op() to get proper behavior. When using with diff, we will never say that something is an UPSERT, only INSERT or UPDATE."
> 
> For the composite key piece, I think its still worth adding a note that changes to a column that is part of the composite key will look like a delete, then an insert (this is true for all changefeeds but I think worth pointing out here as well)

@amruss We have this on the CDC queries page currently:

> Returns a string describing the type of event. If a changefeed is running with the [diff](https://www.cockroachlabs.com/docs/v23.1/create-changefeed#diff-opt) option, then this function returns 'insert', 'update', or 'delete'. If a changefeed is running without the diff option, it is not possible to determine an update from an insert, so event_op() returns ['upsert'](https://www.cockroachlabs.com/blog/sql-upsert/) or 'delete'.

**Comment by kathancox** [comment]
> For the composite key piece, I think its still worth adding a note that changes to a column that is part of the composite key will look like a delete, then an insert (this is true for all changefeeds but I think worth pointing out here as well)

@amruss Got it! I misinterpreted Yevgeniy's PR and thought this was somehow resolved. I will add a note for this in now!

File: v23.1/create-changefeed.md [hunk 1]
```diff
@@ -159,7 +159,7 @@ Option | Value | Description
...
-`envelope` | `key_only` / `row`* / `wrapped` | `key_only` emits only the key and no value, which is faster if you only want to know when the key changes.<br><br>`row` emits the row without any additional metadata fields in the message. *You can only use `row` with Kafka sinks or sinkless changefeeds. `row` does not support [`avro` format](#format).<br><br>`wrapped` emits the full message including any metadata fields. See [Responses](changefeed-messages.html#responses) for more detail on message format.<br><br>Default: `envelope=wrapped`
+<a name="envelope"></a>`envelope` | `key_only` / `row`* / `wrapped` | `key_only` emits only the key and no value, which is faster if you only want to know when the key changes.<br><br>`row` emits the row without any additional metadata fields in the message. *You can only use `row` with Kafka sinks or sinkless changefeeds. `row` does not support [`avro` format](#format).<br><br>`wrapped` emits the full message including any metadata fields. See [Responses](changefeed-messages.html#responses) for more detail on message format.<br><br>Default: `envelope=wrapped`
```
Comment (taroface) [inline v23.1/create-changefeed.md:162]: I haven't seen this use of an asterisk elsewhere in our docs. The usual way to emphasize is clearer IMO:

```suggestion
<a name="envelope"></a>`envelope` | `key_only` / `row` / `wrapped` | `key_only` emits only the key and no value, which is faster if you only want to know when the key changes.<br><br>`row` emits the row without any additional metadata fields in the message. **Note:** You can only use `row` with Kafka sinks or sinkless changefeeds. `row` does not support [`avro` format](#format).<br><br>`wrapped` emits the full message including any metadata fields. See [Responses](changefeed-messages.html#responses) for more detail on message format.<br><br>Default: `envelope=wrapped`
```

Comment (taroface) [inline v23.1/create-changefeed.md:162]: I haven't seen this use of an asterisk elsewhere in our docs. The usual way to emphasize is clearer IMO:

```suggestion
<a name="envelope"></a>`envelope` | `key_only` / `row` / `wrapped` | `key_only` emits only the key and no value, which is faster if you only want to know when the key changes.<br><br>`row` emits the row without any additional metadata fields in the message. **Note:** You can only use `row` with Kafka sinks or sinkless changefeeds. `row` does not support [`avro` format](#format).<br><br>`wrapped` emits the full message including any metadata fields. See [Responses](changefeed-messages.html#responses) for more detail on message format.<br><br>Default: `envelope=wrapped`
```

File: _includes/v23.1/cdc/composite-key-delete-insert.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{{site.data.alerts.callout_info}}
+When changes happen to a column that is part of a composite [key](primary-key.html), the changefeed will produce a {% if page.name == "cdc-queries.md" %} [delete message](#filter-delete-messages) {% else %} [delete message](changefeed-messages.html#delete-messages) {% endif %} then an insert message.
```
[the remark this answers is not in the document]
Comment (kathancox) [reply _includes/v23.1/cdc/composite-key-delete-insert.md]: Done

File: _includes/v23.1/known-limitations/cdc-queries.md [hunk 3]
```diff
@@ -5,4 +5,4 @@
     - [Volatile functions](functions-and-operators.html#function-volatility)
```
[the remark this answers is not in the document]
Comment (kathancox) [reply _includes/v23.1/known-limitations/cdc-queries.md]: Done!

File: v23.1/cdc-queries.md [hunk 4]
```diff
@@ -138,6 +140,23 @@ CREATE CHANGEFEED INTO sink AS SELECT * FROM table WHERE NOT event_op() = 'delet
...
+### Capture delete messages
+
+Delete changefeed messages will only contain the primary key value and all other columns will emit as `NULL` (see the [Limitations](#limitations)). To emit the deleted values use the [`envelope=wrapped`](create-changefeed.html#envelope), [`format=json`](create-changefeed.html#format), and [`diff`](create-changefeed.html#diff-opt) options:
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+CREATE CHANGEFEED INTO 'external://cloud' WITH envelope='wrapped', format='json', diff AS SELECT * FROM users WHERE event_op() = 'delete';
+~~~
+
+This will produce a JSON object with `before` and `after` keys that contain the prior and current state of the row:
```
[the remark this answers is not in the document]
Comment (kathancox) [reply v23.1/cdc-queries.md]: Done!

File: v23.1/cdc-queries.md [hunk 5]
```diff
@@ -138,6 +140,23 @@ CREATE CHANGEFEED INTO sink AS SELECT * FROM table WHERE NOT event_op() = 'delet
...
+
+~~~
+{"after": null, "before": {"address": "95913 Thomas Key Apt. 99", "city": "washington dc", "credit_card": "2702281601", "id": "49a8c43d-8ed8-4d50-ad99-fb314cbe20a1", "name": "Tina Jones"}}
+~~~
+
+The `before` value in the message, produced by the `diff` option, will include the entire row. That is, it will not include any [projections](#syntax) from a CDC query.
```
[the remark this answers is not in the document]
Comment (kathancox) [reply v23.1/cdc-queries.md]: Done!

File: v23.1/create-changefeed.md (hunk 1 above)
Comment (kathancox) [reply v23.1/create-changefeed.md]: Yes, good point, this has been here a while. Much better!

**Comment by kathancox** [comment]
Got a LGTM from @miretskiy in Slack

**Review by miretskiy — APPROVED** [review]
Here is a real lgtm too.

**Comment by kathancox** [comment]
> Here is a real lgtm too.

lol thanks Yevgeniy!

**Comment by kathancox** [comment]
> Looks good! Just minor suggestions.

TFTR @taroface

**Pull request merged**